extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionValidateOnMigrateFailureTest.java

```diff
@@ -0,0 +1,31 @@
+package io.quarkus.flyway.test;
+
+import org.flywaydb.core.api.FlywayException;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+/**
+ * Test suite for Flyway's validate.
+ * <p>
+ * Assures Flyway's validate runs and fails consistently when SQL migrations applied do not match
+ * the one(s) available locally.
+ */
+public class FlywayExtensionValidateOnMigrateFailureTest {
+
+    @RegisterExtension
+    final QuarkusUnitTest config = new QuarkusUnitTest().setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+            .addAsResource("db/migration/V1.0.0__Quarkus.sql")
+            .addAsResource("validate-on-migrate-failure-config.properties", "application.properties"))
+            .setExpectedException(FlywayException.class);
+
+    @Test
+    @DisplayName("validate on migrate fails if migrations applied do not match the one(s) available locally")
+    public void testFlywayConfigInjection() {
+        // ...
```

Comment:
Add a fail() if this test is supposed to never be called

Reply:
This one is tricky. The `setExpectedException` call will take of the test, failing when the application boostraps (the correct behavior), but the method is only there because otherwise the class won't execute any test(s).

If you have a better way to test this I'm all ears.